engine/crates/engine/registry-v2/src/common/iter.rs

```diff
@@ -0,0 +1,75 @@
+use std::iter::FusedIterator;
+
+use crate::{
+    common::{IdOperations, IdRange},
+    Registry, RegistryId,
+};
+
+/// Iterator for readers
+///
+/// T indicates the type that will be yielded by the Iterator
+#[derive(Clone, Copy)]
+pub struct Iter<'a, T>
+where
+    T: IdReader,
+{
+    range: IdRange<T::Id>,
+    current: Option<T::Id>,
```

Comment:
any reason to have `current`? sounds only useful to implement `peek()`

Reply:
Probably not.  First pass of this had a(n incorrect) DoubleEndedIterator impl that needed current.  It is pretty useless now that I've realised how you're actually meant to do DoubleEndedIterator.  

But it's also harmless so I might hold off removing it till I'm done with the mountain of other things

Reply:
This was trivial to do, so done